src/sidebar/services/test/annotations-exporter-test.js

```diff
@@ -36,4 +42,84 @@ describe('AnnotationsExporter', () => {
       annotations: [firstBaseAnnotation, secondBaseAnnotation],
     });
   });
+
+  describe('buildTextExportContent', () => {
+    it('throws error when empty list of annotations is provided', () => {
+      assert.throws(
+        () => exporter.buildTextExportContent([]),
+        'No annotations to export',
+      );
+    });
+
+    it('generates text content with provided annotations', () => {
+      const isoDate = now.toISOString();
+      const annotation = {
+        ...newAnnotation(),
+        ...publicAnnotation(),
+        created: isoDate,
+      };
+      // Title should actually be an array
+      annotation.document.title = [annotation.document.title];
+      const annotations = [
+        annotation,
+        annotation,
+        {
+          ...annotation,
+          user: 'acct:jane@localhost',
+          tags: ['foo', 'bar'],
+        },
+        {
+          ...annotation,
+          ...newReply(),
+        },
+      ];
+      const groupName = 'My group';
+
+      const result = exporter.buildTextExportContent(
+        annotations,
+        groupName,
+        now,
+      );
+
+      assert.equal(
+        result,
+        `${isoDate}
+A special document
+http://example.com
+Group: ${groupName}
+Total users: 2
+Users: bill, jane
+Total annotations: 4
+Total replies: 1
+
+Annotation 1:
+${isoDate}
+null
+bill
+"Annotation text"
+Tags: [tag_1, tag_2]
+
+Annotation 2:
+${isoDate}
+null
+bill
+"Annotation text"
+Tags: [tag_1, tag_2]
+
+Annotation 3:
+${isoDate}
+null
+jane
+"Annotation text"
+Tags: [foo, bar]
+
+Annotation 4:
+${isoDate}
+null
+bill
+"Annotation text"
+Tags: [tag_1, tag_2]`,
```

Comment:
I was tempted to use `trimAndDedent` here, to make the result a bit more legible, but then we would not be testing potential flaws on that particular function, so even though this is uglier, it'e better to assert against an expected raw result.